providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -204,6 +205,12 @@ private <T> ProviderEvaluation<T> resolve(Class<T> type, String key, EvaluationC
         // check variant existence
         Object value = flag.getVariants().get(resolvedVariant);
         if (value == null) {
+            if (resolvedVariant.isEmpty()) {
+                String message = String.format("no default variant found in flag with key %s", key);
+                log.debug(message);
+                throw new FlagNotFoundError(message);
```

Comment:
```suggestion
                return ProviderEvaluation.<T>builder()
                        .reason(Reason.ERROR.toString())
                        .errorCode(ErrorCode.FLAG_NOT_FOUND)
                        .errorMessage("Flag '" + key+ "' has no default variant defined, will use code default")
                        .flagMetadata(getFlagMetadata(storageQueryResult))
                        .build();
```
most likely this will do the trick and is in sync with the js implementation